test/options_test.cc

```diff
@@ -272,10 +273,115 @@ TEST_F(OptionsImplTest, RequestSource) {
   // Check that our conversion to CommandLineOptionsPtr makes sense.
   CommandLineOptionsPtr cmd = options->toCommandLineOptions();
   EXPECT_EQ(cmd->request_source().uri(), request_source);
+  // TODO(#433)
   OptionsImpl options_from_proto(*cmd);
   EXPECT_TRUE(util(*(options_from_proto.toCommandLineOptions()), *cmd));
 }
 
+class RequestSourcePluginTestFixture : public OptionsImplTest,
+                                       public WithParamInterface<std::string> {};
+
+TEST_P(RequestSourcePluginTestFixture, CreatesOptionsImplWithRequestSourceConfig) {
+  Envoy::MessageUtil util;
+  const std::string request_source_config = GetParam();
+  std::unique_ptr<OptionsImpl> options = TestUtility::createOptionsImpl(
+      fmt::format("{} --request-source-plugin-config {} {}", client_name_, request_source_config,
+                  good_test_uri_));
+
+  CommandLineOptionsPtr command = options->toCommandLineOptions();
+  EXPECT_TRUE(
+      util(command->request_source_plugin_config(), options->requestSourcePluginConfig().value()));
+  // Now we construct a new options from the proto we created above. This should result in an
+  // OptionsImpl instance equivalent to options. We test that by converting both to yaml strings,
+  // expecting them to be equal. This should provide helpful output when the test fails by showing
+  // the unexpected (yaml) diff.
+
+  // The predicates are defined as proto maps, and these seem to re-serialize into a different
+  // order. Hence we trim the maps to contain a single entry so they don't thwart our textual
+  // comparison below.
+  EXPECT_EQ(1, command->mutable_failure_predicates()->erase("benchmark.http_4xx"));
+  EXPECT_EQ(1, command->mutable_failure_predicates()->erase("benchmark.http_5xx"));
+  EXPECT_EQ(1, command->mutable_failure_predicates()->erase("requestsource.upstream_rq_5xx"));
+  // TODO(#433)
+  OptionsImpl options_from_proto(*command);
+  std::string s1 = Envoy::MessageUtil::getYamlStringFromMessage(
+      *(options_from_proto.toCommandLineOptions()), true, true);
+  std::string s2 = Envoy::MessageUtil::getYamlStringFromMessage(*command, true, true);
+  EXPECT_EQ(s1, s2);
+  // Additional comparison to avoid edge cases missed.
+  EXPECT_TRUE(util(*(options_from_proto.toCommandLineOptions()), *command));
+}
+
+INSTANTIATE_TEST_SUITE_P(
+    HappyPathRequestSourceConfigJsonSuccessfullyTranslatesIntoOptions,
+    RequestSourcePluginTestFixture,
+    ::testing::ValuesIn(std::vector<std::string>{
```

Comment:
I'm having a really hard time parsing this.  There are two things that might help here:
1. C++ has a formatted string option that will make this much easier to read, rather than using \" so much. You can do R"( string that contains " with no problems)". https://abseil.io/tips/64
2. (optional) I _think_ this is just one string, but it's difficult to tell.  Can we abstract the actual textprotos into variables?

Reply:
I'm not sure that 1 works for me here. I might be missing something, but it seems like there are issues with the newlines. At least it seems like using R"( makes you keep the newlines if you space them into separate lines. I can make each line a separate raw string though, I'm not sure that this makes it more readable, but maybe you'll find this preferable?
2. I've kind of tried to do this. Hope that this approach looks ok.